share/random_shares.go

```diff
@@ -0,0 +1,69 @@
+package share
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"sort"
+)
+
+// RandShares generates total amount of shares and fills them with random data.
+func RandShares(total int) []Share {
+	if total&(total-1) != 0 {
+		panic(fmt.Errorf("total must be power of 2: %d", total))
```

Comment:
[optional] return an error instead of `panic`ing. 

Tangent: it seems like a weird constraint that `RandShares` must be invoked with a power of `2`.